app/controllers/api/v2/application_controller.rb

```diff
@@ -5,23 +5,35 @@ class ApplicationController < ::ApplicationController
 
       def authenticate
         authenticate_or_request_with_http_token do |token|
-          if Settings.authentication.algorithm == 'plain-text'
-            # This method can be used in development mode to simplify querying
-            # the API with curl. It should allow us to do:
-            #
-            #    curl -H 'Authorization: Bearer user@education.gov.uk' http://localhost:3001/api/v2/providers
-            email = token
-          else
-            (payload, _options) = JWT.decode(token,
-                                                 Settings.authentication.secret,
-                                                 Settings.authentication.algorithm)
-            email = payload['email']
-          end
-          # match email addresses case-insensitively
+          email = get_email_from_token(token)
+
           @current_user = User.find_by("lower(email) = ?", email.downcase)
+
+          if @current_user.present? && @current_user.accept_terms_date_utc.nil?
+            head 403
+          end
+
           @current_user.present?
         end
       end
+
+    private
+
+      def get_email_from_token(token)
+        if Settings.authentication.algorithm == 'plain-text'
+          # This method can be used in development mode to simplify querying
+          # the API with curl. It should allow us to do:
+          #
+          #    curl -H 'Authorization: Bearer user@education.gov.uk' http://localhost:3001/api/v2/providers
+          email = token
+        else
+          (payload, _options) = JWT.decode(token,
+                                               Settings.authentication.secret,
+                                               Settings.authentication.algorithm)
+          email = payload['email']
+        end
+        email
```

Comment:
is this line really needed?

i thought that if blocks return and else block returns you get the email anyway

Reply:
It's not, but I think it's a bit more explicit... not fussed with either way though.

Reply:
you could do

```ruby
def get_email_from_token(token)
  if Settings.authentication.algorithm == 'plain-text'
    # ...
    token
  else
    ...
    payload['email']
  end
end
```